src/app/meta-campaign/meta-campaign.component.ts

```diff
@@ -97,7 +103,11 @@ export class MetaCampaignComponent implements OnInit {
     }
   }
 
-  setDefaultFilters() {
+  /**
+   * @method  setDefaultFilters
+   * @param   setQueryParams determine whether to override query with params in the URL
+   */
+  setDefaultFilters(setQueryParams: boolean) {
```

Comment:
Why wouldn't we just always set the query params?

Reply:
@NoelLH When `onClearFiltersApplied()` is triggered I've reused `setDefaultFilters()` to clear all sort / filters - and within `setDefaultFilters()` if `setQueryParams()` is triggered every time, the subscribed `route.queryParams` is overriding the clear method and essentially nothing is emptied.

Reply:
This sounds like something that should be fixed elsewhere then @tbg-jasonfung. Stuff you've done on the page should take precedence if these two things diverge. Moreover, they should not ever diverge because updating the form should immediately update the route, just like other navigation events.

Reply:
I've added a new method called `setDefaultFiltersOrParams()` which helps identify exactly its purpose when query params exist. This gets rid of the Boolean flag and also resolves the filters not clearing.